# Python Insights - Analyzing Data with Python
## Case - Reasons for Cancellations Analysis

Libraries:
Pandas

# Step 0 - Understand the challenge and the company

You've been hired by a company with over 800,000 customers for a Data project. Recently, the company realized that of its total customer base, most are inactive customers, that is, customers who have already canceled the service.

Needing to improve her results, she wants to be able to understand the main reasons for these cancellations and what are the most efficient actions to reduce this number.

In [56]:
# Step 1 - Importing the data base
import pandas
tabela = pandas.read_csv("cancelamentos.csv")

In [ ]:
# Step 2 - Visualization of the data base
display(tabela)

In [ ]:
# Step 3 - Data Cleaning | error handling
# 3.1. - Null Values
tabela = tabela.dropna()
# 3.2. - Unnecessary columns
tabela = tabela.drop("CustomerID",axis=1)
#
display(tabela.info())

In [ ]:
# Step 4 - Explore Data
display(tabela["cancelou"].value_counts(normalize=True).map("{:.1%}".format))
display(tabela["duracao_contrato"].value_counts(normalize=True).map("{:.1%}".format))
display(tabela.groupby("duracao_contrato").count())
# display(tabela.groupby("duracao_contrato").mean())
# display(tabela.groupby("assinatura").mean())
# for column in tabela.columns:
#   graphic = px.histogram(tabela, x=column,color="cancelou")
#   graphic.show()


In [ ]:
# Step 4.1. - Report insights | Main reasons for cancellations
import plotly.express as px

# a. 100% of monthly contract customers cancelled the subscription
graphic = px.histogram(tabela, x="duracao_contrato",color="cancelou")
graphic.show()

# b. 100% of the customers with 50+ cancelled the subscription
graphic = px.histogram(tabela, x="idade",color="cancelou")
graphic.show()

# c. 100% of the customers that called more than 5x, cancelled the subscription
graphic = px.histogram(tabela, x="ligacoes_callcenter",color="cancelou")
graphic.show()

# d. 100% of the customers who are overdue for more than 20 days cancelled their subscription
graphic = px.histogram(tabela, x="dias_atraso",color="cancelou")
graphic.show()

In [57]:
# Step 5 - Proposed solutions
display(tabela["cancelou"].value_counts(normalize=True).map("{:.1%}".format))

# a. Exclude Monthly Payment Method or Offer Discounts
tabela = tabela[tabela["duracao_contrato"]!="Monthly"]
display(tabela["cancelou"].value_counts(normalize=True).map("{:.1%}".format))

# b. Targeted Engagement for Customers Over 50 Years
tabela = tabela[tabela["idade"]<50]
display(tabela["cancelou"].value_counts(normalize=True).map("{:.1%}".format))

# c. Enhance Call Center Support
tabela = tabela[tabela["ligacoes_callcenter"]<5]
display(tabela["cancelou"].value_counts(normalize=True).map("{:.1%}".format))

# d. Proactive Payment Monitoring
tabela = tabela[tabela["dias_atraso"]<=20]
display(tabela["cancelou"].value_counts(normalize=True).map("{:.1%}".format))


cancelou
1.0    56.7%
0.0    43.3%
Name: proportion, dtype: object

cancelou
0.0    53.9%
1.0    46.1%
Name: proportion, dtype: object

cancelou
0.0    63.3%
1.0    36.7%
Name: proportion, dtype: object

cancelou
0.0    81.2%
1.0    18.8%
Name: proportion, dtype: object

cancelou
0.0    87.8%
1.0    12.2%
Name: proportion, dtype: object

By implementing targeted solutions, including excluding the monthly payment method or offering discounts, implementing targeted engagement for customers over 50 years, enhancing call center support, and proactive payment monitoring, our company can reduce the cancellation rate from 56% to an impressive 12%